synapse/storage/data_stores/main/room.py

```diff
@@ -98,6 +98,53 @@ def get_room(self, room_id):
             allow_none=True,
         )
 
+    def get_room_stats_state(self, room_id):
+        """Retrieve room stats state.
+
+        Args:
+            room_id (str): The ID of the room to retrieve.
+        Returns:
+            A dict containing the room information, or None if the room is unknown.
+        """
+        return self.db.simple_select_one(
+            table="room_stats_state",
+            keyvalues={"room_id": room_id},
+            retcols=(
+                "room_id",
+                "name",
+                "canonical_alias",
+                "encryption",
+                "is_federatable",
+                "join_rules",
+                "guest_access",
+                "history_visibility",
+                "topic",
+            ),
+            desc="get_room_stats_state",
+            allow_none=True,
+        )
+
+    def get_room_stats_current(self, room_id):
+        """Retrieve current room stats.
+
+        Args:
+            room_id (str): The ID of the room to retrieve.
```

Comment:
Please use type hints in the function definition instead of in the docstring. (This also applies to `get_room_stats_state` above.)